File: Frontend/types/notification.ts
// src/types/notification.ts
export interface Notification {
  id: number;
  type: 'SUCCESS' | 'WARNING' | 'INFO' | 'ERROR';
  titre: string;
  message: string;
  details: string | null;
  lien: string | null;
  lu: boolean;
  dateCreation: string;
  tempsRelatif?: string;
}

export interface PageResponse<T> {
  content: T[];
  totalElements: number;
  totalPages: number;
  size: number;
  number: number;
}

File: Backend/sigr/src/main/java/com/alomrane/sigr/model/Notification.java
package com.alomrane.sigr.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "notifications")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Notification {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "utilisateur_id", nullable = false)
    private Utilisateur utilisateur;

    @Enumerated(EnumType.STRING)
    private TypeNotification type; // SUCCESS, WARNING, INFO, ERROR

    private String titre;
    private String message;
    private String details; // optionnel (texte long)

    private String lien; // URL relative (ex: "/dashboard/suivi?id=123")

    private boolean lu;

    private LocalDateTime dateCreation;

    @PrePersist
    protected void onCreate() {
        dateCreation = LocalDateTime.now();
    }

    public enum TypeNotification {
        SUCCESS, WARNING, INFO, ERROR
    }
}